### PR TITLE
Assembly toCompound

### LINKED ISSUE #629: There is no function to create a solid from assembly
Creating a solid from assembly has following use cases:

- Exporting the assembly to format that does not support multiple parts, for example stl
- Using assembly in boolean operations

### LINKED ISSUE #724: Bounding box for an assembly
When positioning subassemblies within a parent assembly it would be useful to get the bounding box of the (already solved) subassembly.

Possibly creating a Compound from an Assembly could be an easy way to implement, if so then this issue is related to #629.

### PR DESCRIPTION
**Pull request by marcus7070**
Will resolve #724 (since you can now do `assy.toCompound().BoundingBox()`), and I think it's the best way to resolve #629.

### CODE REVIEW

File: tests/test_assembly.py [hunk 1]
```diff
@@ -391,3 +391,42 @@ def resulting_plane(shape0):
...
+    c0 = simple_assy.toCompound()
+    assert isinstance(c0, cq.Compound)
+    assert len(c0.Solids()) == 4
+
+    c1 = nested_assy.toCompound()
+    assert isinstance(c1, cq.Compound)
+    assert len(c1.Solids()) == 4
+
+    # check nested assy location appears in compound
+    # create four boxes, stack them ontop of each other, check highest face is in final compound
```
Comment (jmwright) [inline tests/test_assembly.py:407]:
```suggestion
    # create four boxes, stack them on top of each other, check highest face is in final compound
```

**Review by jmwright — APPROVED** [review]
Thanks!

**Review by adam-urbanczyk — APPROVED** [review]

**Comment by jmwright** [comment]
Looks like there's a mypy error.

Screenshot from 2021-04-19 06-07-24

**Comment by adam-urbanczyk** [comment]
Fixed, @jmwright. Shall I merge?

**Comment by jmwright** [comment]
+1

**Comment by adam-urbanczyk** [comment]
Thanks @marcus7070 !

**Pull request merged**